source/common/http/conn_manager_impl.cc

```diff
@@ -1288,6 +1288,27 @@ void ConnectionManagerImpl::ActiveStream::refreshCachedRoute() {
         connection_manager_.cluster_manager_.get(stream_info_.route_entry_->clusterName());
     cached_cluster_info_ = (nullptr == local_cluster) ? nullptr : local_cluster->info();
   }
+  refreshCachedTracingCustomTags();
+}
+
+void ConnectionManagerImpl::ActiveStream::refreshCachedTracingCustomTags() {
+  std::map<absl::string_view, Tracing::CustomTagPtr> merged;
```

Comment:
why do you need convert between map <-> vector twice? If you keep use map every where then you can use absl::flat_hash_map::merge for this logic.

Reply:
Thanks for the suggestion, i'll fix it.

Reply:
It's not allowed to merge directly i think. Seems it would erase the source, it's a move action. A copy needed or other options.